lms/static/scripts/frontend_apps/components/test/AuthorizationModal-test.js

```diff
@@ -0,0 +1,48 @@
+import { mount } from 'enzyme';
+
+import AuthorizationModal from '../AuthorizationModal';
+
+describe('AuthorizationModal', () => {
+  let fakeOnAuthComplete;
+
+  const createComponent = (props = {}) =>
+    mount(
+      <AuthorizationModal
+        authToken="123456"
+        authURL="http://what.evs"
+        onAuthComplete={fakeOnAuthComplete}
+        {...props}
+      />
+    );
+
+  beforeEach(() => {
+    fakeOnAuthComplete = sinon.stub();
+  });
+
+  it('provides an authorization button', () => {
```

Comment:
These two tests are testing a combination of `AuthorizationModal`, `AuthButton` and `ErrorModal`. Since `AuthorizationModal` is very simple, there isn't a huge amount of value in a pure unit test for it. Maybe this is a reasonable thing to do for the moment. This might cause some confusion in future for someone adding to this code - should they be testing just this component or the component and its dependencies.

The alternative would be to have just a very simple test that checks this component renders an `ErrorModal` with the expected buttons. The more valuable testing would then come via integration tests for the overall assignment configuration app.

Reply:
Hey, I am glad you caught this. When I was writing tests I kept thinking "I'm really just writing this test to ensure coverage, but I'm meta-testing, really." It felt awkward and forced. Note that to pull this off, I had to _not_ mock components we would normally mock. Do you have any more specific suggestions for the time being?

I also agree that integration testing here is more valuable.